include/rocksdb/io_status.h

```diff
@@ -28,7 +28,7 @@ class IOStatus : public Status {
   using Code = Status::Code;
   using SubCode = Status::SubCode;
 
-  enum IOErrorScope {
+  enum IOErrorScope : unsigned char {
     kIOErrorScopeFileSystem,
```

Comment:
Is it intentional that ScopeFileSystem is the same as not setting a scope (both 0)?

Reply:
Yes. Unless the scope is explicitly set, we take a conservative approach. I'm trying to make sure that there's no behavior change with this PR.